### PR TITLE
SLS-582 Configuration for delta generation

### PR DESCRIPTION
**Pull request by korteland**
This adds tree-level configuration for the "20% of a page size" heuristic that delta generation previously used. It allows values over 100%, with the idea that if there are cases where deltas end up larger than the page image, we can still play with that and do measurements.

It also includes a test change intended to more aggressively generate deltas.

### CODE REVIEW

**Review by ddanderson — APPROVED** [review]
This looks good, and it will be nice for testing.
I have a couple little design thoughts to consider, neither is a big deal to me.

File: dist/api_data.py [hunk 1]
```diff
@@ -340,6 +340,14 @@ def __ge__(self, other):
         \c unencrypted setting only checksums blocks that are not encrypted. See @ref
         tune_checksum for more information.''',
         choices=['on', 'off', 'uncompressed', 'unencrypted']),
+    Config('delta_pct', '20', r'''
+        the size threshold (as a percentage) at which a delta will cease to be emitted when
+        reconciling a page. For example, if this is set to 20, the size of a delta is 19 bytes,
+        and the size of the full page image is 100 bytes, reconciliation can emit a delta for the
```
[the remark this answers is not in the document]
Comment (pmacko86) [reply dist/api_data.py:346]: I also agree that <= 20% would be a more natural semantic than < 20%, but either could work.

File: test/suite/test_oligarch09.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ class test_oligarch09(wttest.WiredTigerTestCase, DisaggConfigMixin):
     nitems = 1000
 
     def session_create_config(self):
-        cfg = 'key_format=S,value_format=S,block_compressor={}'.format(self.block_compress)
+        cfg = 'delta_pct=200,key_format=S,value_format=S,block_compressor={}'.format(self.block_compress)
```
Comment (pmacko86) [inline test/suite/test_oligarch09.py:63]: Maybe add a comment explaining the value 200, as I'm assuming you'd like to allow deltas larger than a page here? Without it, "200%" looks like a mistake.

**Review by pmacko86 — APPROVED** [review]
LGTM, with minor comments.

File: dist/api_data.py [hunk 3]
```diff
@@ -340,6 +340,14 @@ def __ge__(self, other):
         \c unencrypted setting only checksums blocks that are not encrypted. See @ref
         tune_checksum for more information.''',
         choices=['on', 'off', 'uncompressed', 'unencrypted']),
+    Config('delta_pct', '20', r'''
+        the size threshold (as a percentage) at which a delta will cease to be emitted when
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply dist/api_data.py:344]: Do we want to allow btrees to have different delta percent?

[the remark this answers is not in the document]
Comment (korteland) [reply dist/api_data.py:344]: My original design was a connection-level setting. The problem is that it's for experimentation (at least initially), and I could easily see us wanting to try different values for (say) indexes. So I moved it to the B-tree layer, but there isn't a top-level `disagg` config item there.

I could make one, but the interaction with the `block_manager` config needs some design work. (Do we really want `disagg=(block_manager=disagg)`?)

I think a ticket for adding this top-level config would be the best way forwards.

**Review by quchenhao — APPROVED** [review]
lgtm

File: dist/api_data.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (korteland) [reply dist/api_data.py:344]: Addendum: the other part of it is that there's already server code using the `block_manager` config, and I don't want to rock that boat while we're behind on vendoring. 

What do you think about adding a btree-level `disagg` config with just `delta_pct`, and leaving the `block_mananger` bit for later?

[the remark this answers is not in the document]
Comment (quchenhao) [reply dist/api_data.py:344]: Yes, I think that is good.

[the remark this answers is not in the document]
Comment (korteland) [reply dist/api_data.py]: Done in 90005a3

File: dist/api_data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (korteland) [reply dist/api_data.py]: Good call, donein 90005a3.

File: test/suite/test_oligarch09.py (hunk 2 above)
Comment (korteland) [reply test/suite/test_oligarch09.py]: You're right, added a comment in 90005a3.

**Pull request merged**